src/test/java/uk/gov/hmcts/reform/migration/service/MigrateChildrenServiceTest.java

```diff
@@ -0,0 +1,463 @@
+package uk.gov.hmcts.reform.migration.service;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import uk.gov.hmcts.reform.ccd.client.model.CaseDetails;
+import uk.gov.hmcts.reform.domain.common.CollectionEntry;
+import uk.gov.hmcts.reform.fpl.domain.CaseData;
+import uk.gov.hmcts.reform.fpl.domain.OldChild;
+import uk.gov.hmcts.reform.fpl.domain.OldChildren;
+import uk.gov.hmcts.reform.fpl.domain.Child;
+import uk.gov.hmcts.reform.domain.common.Address;
+import uk.gov.hmcts.reform.domain.common.EmailAddress;
+import uk.gov.hmcts.reform.domain.common.TelephoneNumber;
+import uk.gov.hmcts.reform.fpl.domain.common.ChildParty;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+@SpringBootTest
+class MigrateChildrenServiceTest {
+
+    @Autowired
+    private ChildrenDataMigrationService service;
+
+    @Test
+    void whenOldStructureExistsAcceptsShouldReturnTrue() {
+        Map<String, Object> data = new HashMap<>();
+
+        OldChild child = OldChild.builder().build();
+
+        data.put("children", child);
+
+        CaseDetails caseDetails = CaseDetails.builder()
+            .id(1111L)
+            .data(data)
+            .build();
+
+        assertThat(service.accepts().test(caseDetails)).isEqualTo(true);
+    }
+
+    @Test
+    void whenOldStructureDoesNotExistAcceptsShouldReturnFalse() {
+        CaseDetails caseDetails = CaseDetails.builder()
+            .id(1111L)
+            .data(ImmutableMap.of("data", "someData"))
+            .build();
+
+        assertThat(service.accepts().test(caseDetails)).isEqualTo(false);
+    }
+
+    @Test
+    void whenDataIsNullAcceptsShouldReturnFalse() {
+        CaseDetails caseDetails = CaseDetails.builder()
+            .id(1111L)
+            .data(null)
+            .build();
+
+        assertThat(service.accepts().test(caseDetails)).isEqualTo(false);
+    }
+
+    @SuppressWarnings({"unchecked", "LineLength"})
+    @Test
+    void whenOldChildrenStructureIsMigratedShouldReturnNewListStructure() {
+        CaseData migratedCaseData = service.migrate(createOldChild(false, false));
```

Comment:
Not ideal design, but for one time tool it's fine, false false is a bit confusing (although it displays a bit nicer in the IntelliJ!)

Reply:
I am not inclined to change due to this all being throw away code, it works for now and it proves that we are trying to do is working 👍